wremnants/CardTool.py

```diff
@@ -862,13 +878,22 @@ def fillCardWithSyst(self, syst):
         systnamesPruned = [s for s in systNames if not self.isExcludedNuisance(s)]
         systNames = systnamesPruned[:]
         for chan in self.channels:
-            for systname in systNames:
-                shape = "shape" if not systInfo["noConstraint"] else "shapeNoConstraint"
-                # do not write systs which should only apply to other charge, to simplify card
-                self.cardContent[chan] += f"{systname.ljust(self.spacing)} {shape.ljust(self.systTypeSpacing)} {''.join(include_chan[chan])}\n"
+            systNamesChan = [x.replace("CHANNEL",chan) for x in systNames]
+            for systname in systNamesChan:
+                systShape = shape
+                include_line = include_chan[chan]
+                if systInfo["customizeNuisanceAttributes"]:
```

Comment:
ok, so what could I do.
In all honesty, I've always rooted for a script that simply writes the card, with loops to write all systematics you want for any given species, and all the necessary freedom to modify lines without caring about the original content or definition of the boost histogram. It would be much easier to customize things.
Is the txt datacard  file still needed for the hdf5 output mode?

Reply:
No, in hdf5 output mode the txt datacard doesn't exist. (the corresponding information is contained rather in the metadata inside the hdf5 file, in the same format currently written out by text2hdf5)

Reply:
It's probably not clear if we need this function in the future, so from my side, it is ok to leave it like it is for now. But if the functionality becomes important we may have to change it again.

Reply:
ok